src/pwned_passwords_django/api.py

```diff
@@ -13,42 +14,86 @@
 
 API_ENDPOINT = 'https://api.pwnedpasswords.com/range/{}'
 REQUEST_TIMEOUT = 1.0  # 1 second
+DEFAULT_CACHE_TIMEOUT = 60 * 60  # 1 hour
 USER_AGENT = 'pwned-passwords-django/{} (Python/{} | requests/{})'.format(
     __version__,
     '{}.{}.{}'.format(*sys.version_info[:3]),
     requests.__version__
 )
 
 
-def pwned_password(password):
+def hash_password(password):
     """
-    Checks a password against the Pwned Passwords database.
+    Get the SHA-1 k-anonymity prefix and suffix of a password.
 
     """
     if not isinstance(password, text_type):
         raise TypeError('Password values to check must be Unicode strings.')
     password_hash = hashlib.sha1(password.encode('utf-8')).hexdigest().upper()
     prefix, suffix = password_hash[:5], password_hash[5:]
+    return prefix, suffix
+
+
+def get_pwned_list(prefix):
+    """
+    Get the list of all pwned password hashes for a given SHA-1 prefix.
+
+    """
     try:
-        results = [
-            int(line.partition(':')[2])
-            for line in requests.get(
-                    url=API_ENDPOINT.format(prefix),
-                    headers={'User-Agent': USER_AGENT},
-                    timeout=getattr(
-                        settings,
-                        'PWNED_PASSWORDS_API_TIMEOUT',
-                        REQUEST_TIMEOUT
-                    ),
-            ).text.splitlines()
-            if line.startswith(suffix)
-        ]
-    except (requests.RequestException, ValueError) as e:
+        response = requests.get(
+            url=API_ENDPOINT.format(prefix),
+            headers={'User-Agent': USER_AGENT},
+            timeout=getattr(
+                settings,
+                'PWNED_PASSWORDS_API_TIMEOUT',
+                REQUEST_TIMEOUT
+            ),
+        )
+        response.raise_for_status()
+    except requests.RequestException as e:
         # Gracefully handle timeouts and HTTP error response codes.
         log.warning(
             'Skipped Pwned Passwords check due to error: %r', e
         )
         return None
+    result = {}
+    for line in response.text.splitlines():
+        try:
+            suffix, _, times = line.partition(':')
+            result[suffix] = int(times)
+        except ValueError as e:
```

Comment:
What real life use case this `ValueError` truly guarding? At this stage, we know HIBP has returned a 200 response without timing out. This would seem to be guarding against a HIBP 200 response that doesn't adhere to its own specification. Is there any evidence that this can occur in practice? Personally, if the pwned-passwords-django is using the HIBP API incorrectly, I think an unhandled exception is preferable over quietly assuming a valid, un-compromised password. An error will typically be reported to a database or emailed to the site admins.

Reply:
It was originally there because there was no raise_for_status, and could have been a html page we are looking at.
I have added a raise_for_status like the other pull request, so it should theoretically never happen.
I am happy to remove it if @ubernostrum agrees.

Reply:
Ok. I've implemented this suggestion in PR  #15. If it is approved, it can be used as a foundation for this caching work and the fallback work (#13).